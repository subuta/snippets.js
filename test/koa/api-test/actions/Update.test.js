/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import UpdateTest from 'lib/koa/api-test/actions/Update'

describe('koa/api-test/actions/Update', () => {
  it('should Update koa Update action test', () => {
    expect(format(UpdateTest({
      model: 'user',
      config: {}
    }))).toMatchSnapshot()
  })
})
