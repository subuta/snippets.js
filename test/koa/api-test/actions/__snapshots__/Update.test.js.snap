// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa/api-test/actions/Update should Update koa Update action test 1`] = `
"test('update should update user', async (t) => {
  const {request} = t.context

  // mock jwks
  const token = createToken(privateKey, '123', currentUser)
  jwksEndpoint('http://localhost', [{pub: publicKey, kid: '123'}])

  const response = await request
    .put('/api/users/1')
    .set('Authorization', \`Bearer \${token}\`)
    .send({
      article: {
        title: 'Awesome Redux 2'
      }
    })

  t.is(response.status, 200)
  t.deepEqual(response.body.id, 1)
  t.deepEqual(response.body.title, 'Awesome Redux 2')
  t.deepEqual(response.body.tags[0].label, 'react')
})
"
`;
